src/components/Icon/Icon.tsx

```diff
@@ -237,8 +241,32 @@ function Icon({
   );
 }
 
-function isBundledIcon(key: string | BundledIcon): key is BundledIcon {
-  return Object.keys(BUNDLED_ICONS).includes(key);
+function renderSVG(iconSource: SVGSource) {
+  return (
+    <svg
+      className={styles.Svg}
+      viewBox={iconSource.viewBox}
+      dangerouslySetInnerHTML={{__html: iconSource.body}}
+      focusable="false"
+      aria-hidden="true"
+    />
+  );
+}
+
+function isBundledIcon(key: IconSource): key is BundledIcon {
+  return typeof key === 'string' && Object.keys(BUNDLED_ICONS).includes(key);
+}
+
+function isSVGSource(source: IconSource): source is SVGSource {
+  return (
+    source != null &&
+    source.hasOwnProperty('viewBox') &&
+    source.hasOwnProperty('body')
+  );
+}
+
+function isUntrustedSVG(source: IconSource): source is UntrustedSVG {
```

Comment:
What do you think of moving this check inline instead of in a function? It's a really simple check, so I don't want to inspire undue confidence in it

Reply:
I think keeping it a separate function is okay because it's clear what the conditional is doing through the function name. If we use `typeof source === 'string'` above, it's not as clear what we are testing. The function also ensures source is `UntrustedSVG` though type guarding.